plugins/chain_plugin/test/test_account_query_db.cpp

```diff
@@ -0,0 +1,100 @@
+#define BOOST_TEST_MODULE account_query_db
+#include <eosio/chain/permission_object.hpp>
+#include <boost/test/unit_test.hpp>
+#include <eosio/testing/tester.hpp>
+#include <eosio/chain/types.hpp>
+#include <eosio/chain/block_state.hpp>
+#include <eosio/chain_plugin/account_query_db.hpp>
+
+#ifdef NON_VALIDATING_TEST
+#define TESTER tester
+#else
+#define TESTER validating_tester
+#endif
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::testing;
+using namespace eosio::chain_apis;
+
+using params  = account_query_db::get_accounts_by_authorizers_params;
+using results = account_query_db::get_accounts_by_authorizers_result;
+
+bool find_account_name(results rst, account_name name){
+    for (const auto acc : rst.accounts){
+        if (acc.account_name == name){
+            return true;
+        }
+    }
+    return false;
+}
+bool find_account_auth(results rst, account_name name, permission_name perm){
+    for (const auto acc : rst.accounts){
+        if (acc.account_name == name && acc.permission_name == perm)
+            return true;
+    }
+    return false;
+}
+
+BOOST_AUTO_TEST_SUITE(account_query_db_tests)
+
+BOOST_FIXTURE_TEST_CASE(newaccount_test, TESTER) { try {
+
+	// instantiate an account_query_db
+	auto aq_db = account_query_db(*control);
+
+    produce_blocks(10);
+
+    //link aq_db to the `accepted_block` signal on the controller
+	auto c2 = control->accepted_block.connect([&](const block_state_ptr& blk) {
+        aq_db.commit_block( blk);
+	});
+
+	account_name tester_account = N(tester);
+	const auto tracePtr =  create_account(tester_account);
```

Comment:
Sorry to be that guy, can we make this `trace_ptr`?

Reply:
fixed.